source/s3_request_messages.c

```diff
@@ -266,7 +266,7 @@ struct aws_http_message *aws_s3_create_multipart_upload_message_new(
         false /*exclude_x_amz_meta*/);
 
     if (message == NULL) {
-        goto error_clean_up;
```

Comment:
trivial: I prefer the original implementation. This forces us to not do any allocation before `aws_s3_message_util_copy_http_message_no_body_filter_headers`. We are not doing any right now but it might change in future.